File: backend/app/schemas/mock_exam.py
"""
Mock exam schemas.
"""

from uuid import UUID

from pydantic import BaseModel
from app.schemas.question import QuestionOptionResponse


class MockExamCreate(BaseModel):
    """
    Mock exam creation payload.
    """

    question_count: int = 10
    subject: str | None = None
    topic: str | None = None
    difficulty: int | None = None


class MockExamQuestionResponse(BaseModel):
    """
    Mock exam question response.
    """

    id: UUID

    statement: str

    explanation: str | None = None

    subject: str

    topic: str

    difficulty: int

    options: list[
        QuestionOptionResponse
    ]

    class Config:
        from_attributes = True


class MockExamResponse(BaseModel):
    """
    Mock exam response schema.
    """

    id: UUID
    score: float | None

    class Config:
        from_attributes = True


class SubmitAnswerPayload(BaseModel):
    """
    Submit answer payload.
    """

    question_id: UUID
    selected_option_id: UUID
    response_time: int | None = None

File: frontend/src/types/mock-exam.ts
export interface MockExamResult {
  mockExamId: string;
  score: number;
  correct_answers: number;
  total_answers: number;
  totalTime: number;
}

export interface MockExamOption {
  id: string;
  option_text: string;
}

export interface Question {
  id: string;
  statement: string;
  explanation?: string | null;
  subject: string;
  topic: string;
  difficulty: string;
  options: MockExamOption[];
}

export interface SubmitAnswerPayload {
  question_id: string;
  selected_option_id: string;
  response_time: number;
}

export interface AnsweredQuestion {
  index: number;
  correct: boolean;
}

export interface ReviewItem {
  question_id: string;
  statement: string;
  explanation: string;
  topic: string;
  subject: string;
  difficulty: string;
  selected_option: string | null;
  correct_option: string | null;
  is_correct: boolean;
}

export interface SubmitAnswerInput {
  mockExamId: string;
  payload: SubmitAnswerPayload;
}
